cmd/gapit/trace.go

```diff
@@ -99,12 +96,32 @@ func (verb *traceVerb) Run(ctx context.Context, flags flag.FlagSet) error {
 		}
 
 		if len(devices) == 1 && strings.HasPrefix(traceURI, "port:") {
-			found = append(found, info{
-				uri:    traceURI,
-				device: devices[0],
-				name:   "capture",
-			})
+			target = func(opts *service.TraceOptions) {
+				opts.Device = devices[0]
+				opts.App = &service.TraceOptions_Uri{
+					traceURI,
+				}
+			}
+		} else if len(devices) == 1 && strings.HasPrefix(traceURI, "apk:") {
+			data, err := ioutil.ReadFile(traceURI[4:])
+			if err != nil {
+				return log.Errf(ctx, err, "Failed to read APK at %s", traceURI[4:])
+			}
+			target = func(opts *service.TraceOptions) {
+				opts.Device = devices[0]
+				opts.App = &service.TraceOptions_UploadApplication{
```

Comment:
Does this take care if the apk is already installed?

Reply:
Following this through to here:
https://github.com/google/gapid/blob/master/gapis/trace/android/trace.go#L319

We reinstall the APK if it is already installed (and grant it all permissions)